homeassistant/config_entries.py

```diff
@@ -1525,34 +1526,44 @@ def __setitem__(self, entry_id: str, entry: ConfigEntry) -> None:
         data[entry_id] = entry
         self._index_entry(entry)
 
+    def check_unique_id(self, entry: ConfigEntry) -> None:
+        """Check config entry unique id.
+
+        For hashable return and log warning
+        For non-hashable raise error
+        """
+        unique_id = entry.unique_id
```

Comment:
`if unique_id is None` could be a return guard here to reduce the double check below